app/src/main/java/com/realityexpander/tasky/di/AppModule.kt

```diff
@@ -1,38 +1,103 @@
 package com.realityexpander.tasky.di
 
-import com.realityexpander.tasky.data.repository.AuthRepositoryFakeImpl
-import com.realityexpander.tasky.data.repository.remote.AuthApiFakeImpl
+import com.realityexpander.tasky.BuildConfig
 import com.realityexpander.tasky.data.repository.local.AuthDaoFakeImpl
+import com.realityexpander.tasky.data.repository.authRepositoryImpls.AuthRepositoryFakeImpl
+import com.realityexpander.tasky.data.repository.authRepositoryImpls.AuthRepositoryImpl
+import com.realityexpander.tasky.data.repository.remote.authApiImpls.AuthApiFakeImpl
+import com.realityexpander.tasky.data.repository.local.IAuthDao
+import com.realityexpander.tasky.data.repository.remote.IAuthApi
+import com.realityexpander.tasky.data.repository.remote.authApiImpls.AuthApiImpl
+import com.realityexpander.tasky.data.repository.remote.authApiImpls.TaskyApi
+import com.realityexpander.tasky.data.repository.remote.authApiImpls.TaskyApi.Companion.API_KEY
 import com.realityexpander.tasky.domain.IAuthRepository
 import com.realityexpander.tasky.domain.validation.*
-import com.realityexpander.tasky.domain.validation.validateEmail.EmailMatcherImpl
+import com.realityexpander.tasky.domain.validation.validateEmail.EmailMatcherAndroidImpl
+import com.realityexpander.tasky.domain.validation.validateEmail.EmailMatcherRegexImpl
 import com.realityexpander.tasky.domain.validation.validateEmail.IValidateEmail
 import com.realityexpander.tasky.domain.validation.validateEmail.ValidateEmailImpl
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
 import dagger.hilt.components.SingletonComponent
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.create
+import javax.inject.Named
+import javax.inject.Qualifier
 import javax.inject.Singleton
 
+const val USE_FAKE_REPOSITORY = true
+
 @Module
 @InstallIn(SingletonComponent::class)
 object AppModule {
 
     @Provides
     @Singleton
-    fun provideAuthRepository(): IAuthRepository =
-        AuthRepositoryFakeImpl(
-            authApi = AuthApiFakeImpl(),
-            authDao = AuthDaoFakeImpl(),
-            validateUsername = ValidateUsername(),
-            validateEmail = ValidateEmailImpl(EmailMatcherImpl()),
-            validatePassword = ValidatePassword()
-        )
+    fun provideTaskyApi(): TaskyApi {
+
+        val xApiKeyHeader = Interceptor { chain ->
+            val request = chain.request().newBuilder()
+                .addHeader("x-api-key", API_KEY)
+                .build()
+            chain.proceed(request)
+        }
+
+        val client = if(BuildConfig.BUILD_TYPE == "debug") {
```

Comment:
There's also a field `BuildConfig.DEBUG` which you can directly use for such checks

Reply:
I will use that field instead of checking the build string.